keanu-project/src/test/java/io/improbable/keanu/vertices/VertexLabelTest.java

```diff
@@ -132,44 +132,50 @@ public void youCanDiminishTheNamespace() {
         String innerNamespace = "inner";
         String outerNamespace = "outer";
         String name = "foo";
-        VertexLabel foo = new VertexLabel(name, innerNamespace, outerNamespace);
+        VertexLabel foo = new VertexLabel(outerNamespace, innerNamespace, name);
         VertexLabel newFoo = foo.withoutOuterNamespace();
-        assertThat(newFoo, equalTo(new VertexLabel(name, innerNamespace)));
+        assertThat(newFoo, equalTo(new VertexLabel(innerNamespace, name)));
     }
 
-    @Test(expected = IndexOutOfBoundsException.class)
+    @Test(expected = IllegalArgumentException.class)
```

Comment:
NB you still get an IndexOutOfBoundsException if you do:

        VertexLabel foo = new VertexLabel("foo", "bar").withoutOuterNamespace();
        foo.withoutOuterNamespace();

Reply:
Yeah... the returned list from `withoutOuterNamespace` will be guava so it'll throw `IndexOutOfBoundsException` instead - but I think that's OK since the error message will be self-explanatory. I'll add a test for it though.

Reply:
👍